File: 240515/특정 조건에 맞게 k개 중에 1개를 n번 뽑기/n-permutations-of-k-with-repetition-under-constraint.py
K, N = map(int, input().split())

def back(cnt, sm):
    
    if cnt >= N:
        print(*sm)
        return

    for i in range(1, K+1):
        if len(sm) >= 2:
            if i == sm[-1] == sm[-2]:
                continue
        back(cnt+1, sm+[i])
    
back(0, [])

File: 260428/특정 조건에 맞게 K개 중에 1개를 N번 뽑기/n-permutations-of-k-with-repetition-under-constraint.java
import java.util.*;

public class Main {
    public static int k;
    public static int n;
    public static List<Integer> list = new ArrayList<>();
    public static void main(String[] args) {
       
       Scanner sc = new Scanner(System.in);
       k = sc.nextInt();
       n = sc.nextInt();
       back(1);

    }

    public static void back(int cnt)
    {

        if (cnt > n)
        {
            for (int j = 0; j < list.size(); j++)
            {
                System.out.print(list.get(j) + " ");
            }
            System.out.println();
            return;
        }

        for (int i = 1; i <= k; i++)
        {
                if (cnt >= 3 && i == list.get(list.size()-2) &&  list.get(list.size()-1) ==  list.get(list.size()-2))
                {
                    continue;
                }
                else 
                {
                    list.add(i);
                    back(cnt+1);
                    list.remove(list.size()-1);
                }
        }

    }
}